File: algorithms/[240]搜索二维矩阵 II/solution.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
const searchMatrix = function(matrix, target) {
  if (!matrix.length) return false;
  const m = matrix.length;
  const n = matrix[0].length;

  // 从左下开始搜索
  let [i, j] = [m - 1, 0];
  while (i >= 0 && j < n) {
    if (matrix[i][j] === target) return true;
    if (matrix[i][j] > target) {
      i -= 1;
    } else {
      j += 1;
    }
  }

  return false;
};

File: algorithms/[240]搜索二维矩阵 II/solution.py
from typing import List


class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        i, j = len(matrix) - 1, 0

        while i >= 0 and j < len(matrix[i]):
            val = matrix[i][j]
            if val == target:
                return True
            if val > target:
                i -= 1
            else:
                j += 1

        return False
